### PR TITLE
Use built-in container image registry of ARO 4 cluster

### PR DESCRIPTION
**Pull request by majguo**

### CODE REVIEW

File: articles/openshift/howto-deploy-java-openliberty-app.md [hunk 1]
```diff
@@ -45,41 +45,111 @@ Follow the instructions in these two tutorials and then return here to continue.
...
 
-You can use your ACR instance as a source of container images with your ARO 4 cluster, by creating an image pull secret.
+OpenShift Container Platform provides a built-in container image registry which runs as a standard workload on the ARO 4 cluster. The registry provides an out of the box solution for users to manage the images that run their workloads, and runs on top of the existing cluster infrastructure. Follow the instructions below to enable the access of the built-in registry.
 
-1. Log in to the OpenShift web console from your browser.
+1. Log in to the OpenShift CLI using the token retrieved from the web console with a user from Azure AD.
+   > [!NOTE]
+   > Write down the user name and password for this Azure AD user, as it will be used as **the administrator** for the demo project throughout this guide and subsequent ones.
+2. Run `oc whoami` in the console and write down its output as **\<aad-user>**.
+3. Log in to the OpenShift CLI using the token retrieved from the web console with the `kubeadmin` credentials.
```
Comment (gcharters) [inline articles/openshift/howto-deploy-java-openliberty-app.md]: I found this instruction a little hard to follow.  It may be obvious if I'm actually doing the task, but I wonder if it could be more clearly explained or even a 'template' example of what to type?

**Review by gcharters — COMMENTED** [review]
Just one minor comment/suggestion.  

I did wonder about the use of **<aad-user>** versus **aad-user** but I think I now understand there is an intentional difference.

File: articles/openshift/howto-deploy-java-openliberty-app.md (hunk 1 above)
Comment (gcharters) [reply articles/openshift/howto-deploy-java-openliberty-app.md]: The specific instruction I was referring to is 
> 3. Log in to the OpenShift CLI using the token retrieved from the web console with the `kubeadmin` credentials.

Comment (majguo) [reply articles/openshift/howto-deploy-java-openliberty-app.md]: That's a good point. I pushed the the [following commit](https://github.com/edburns/azure-docs/pull/3/commits/8ed9f264b6a635f949ad9dc9179bb5399ec8e1e6) to address your comments:
- A 'template' example on how to log in to the OpenShift CLI with the token is added as a separate section, **Log in to the OpenShift CLI with the token**;
- Wherever logging in to the OpenShift CLI with the token is required, this section is referenced so user can easily know more details if he/she wants;

**Comment by majguo** [comment]
Hi @gcharters , I also pushed [another commit](https://github.com/edburns/azure-docs/pull/3/commits/febaac39f718fd7b1c6f497f470aff4484974da8) to list minimum cpu cores required for an azure subscription creating an ARO 4 cluster as the 1st prerequisite, per our discussion and agreement reached in the Slack channel.

File: articles/openshift/howto-deploy-java-openliberty-app.md [hunk 2]
```diff
@@ -92,12 +95,14 @@ To access the built-in container image registry provided by the ARO 4 cluster, a
...
 OpenShift Container Platform provides a built-in container image registry which runs as a standard workload on the ARO 4 cluster. The registry provides an out of the box solution for users to manage the images that run their workloads, and runs on top of the existing cluster infrastructure. Follow the instructions below to enable the access of the built-in registry.
 
-1. Log in to the OpenShift CLI using the token retrieved from the web console with a user from Azure AD.
+1. Log in to the OpenShift web console from your browser using the credentials of an Azure AD user.
    > [!NOTE]
    > Write down the user name and password for this Azure AD user, as it will be used as **the administrator** for the demo project throughout this guide and subsequent ones.
-2. Run `oc whoami` in the console and write down its output as **\<aad-user>**.
-3. Log in to the OpenShift CLI using the token retrieved from the web console with the `kubeadmin` credentials.
-4. Execute the following commands to enable the access to the built-in registry for the **aad-user**.
+2. [Log in to the OpenShift CLI with the token](#log-in-to-the-openshift-cli-with-the-token).
```
Comment (gcharters) [inline articles/openshift/howto-deploy-java-openliberty-app.md]: Because item 2 in the list is a link to the instructions for item 1 it creates the impression there are two steps, when actually there's just one.  I would combine these lines to something like:
```
1. Log in to the OpenShift web console from your browser using the credentials of an Azure AD user (for help with how to log in, see [Log in to the OpenShift CLI with the token](#log-in-to-the-openshift-cli-with-the-token) ).
```

I would also do the same for steps 4 and 5.

Comment (majguo) [reply articles/openshift/howto-deploy-java-openliberty-app.md]: Thanks @gcharters for your quick response. Item 1 is to "Log in to the OpenShift **web console**", but item 2 is to "Log in to the OpenShift **CLI** with the token", there're indeed two steps. :)

**Review by gcharters — APPROVED** [review]
Looks good, although I'm not able to test the instructions as I only have a free account.  Thanks.
